pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go

```diff
@@ -363,6 +360,43 @@ func (c *ZookeeperConfig) withDefaults() (changed bool) {
 	return changed
 }
 
+type Persistence struct {
+	// VolumeReclaimPolicy is a zookeeper operator configuration. If it's set to Delete,
+	// the corresponding PVCs will be deleted by the operator when zookeeper cluster is deleted.
+	// The default value is Retain.
+	VolumeReclaimPolicy VolumeReclaimPolicy `json:"reclaimPolicy,omitempty"`
+	// PersistentVolumeClaimSpec is the spec to describe PVC for the container
+	// This field is optional. If no PVC spec, stateful containers will use
+	// emptyDir as volume.
+	PersistentVolumeClaimSpec v1.PersistentVolumeClaimSpec `json:"spec,omitempty"`
+}
+
+func (p *Persistence) withDefaults() (changed bool) {
+	if p.VolumeReclaimPolicy == "" {
```

Comment:
We should check and enforce that the value is one of the accepted ones (i.e. `Delete` or `Retain`). If the value is invalid, we could use the default `Retain` policy.

Reply:
Right! Got it.